### ISSUE TITLE
Aggro during Survival Guide use & teleporting caused black screen

### ISSUE THREAD

**Issue by Aslenta** [DerpyProjectGroup/topaz#37]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/DerpyProjectGroup/topaz/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read the [Contributing Guide](https://github.com/DerpyProjectGroup/topaz/blob/info/CONTRIBUTING.md)
- [x] I have specified what branch this happens on branch: freedom

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 
I got aggro'ed by a goblin when interacting with the Survival Guide in Cape Teriggan, so I quickly selected any location (I think I chose Eastern Adoulin if it matters) to escape. Before getting teleporting out, the screen went black, the goblin was still attacking, and it looked like I could still move judging by the position of the arrow above the target goblin. I couldn't attack the goblin but could open the menu to use my warp ring. The shutdown command worked, but was interrupted when the goblin landed a hit.

**Comment by RAIST5150** [comment]
Likely tied to our known zone out problem in general.

We are still in the zone for a while after crossing the zone line, even during black screen.  If HP is too low, can risk a KO while waiting for the next zone to load.  Have occasionally run into bodies at zone lines because if it.

**Comment by TeoTwawki** [comment]
could have sworn I posted on this issue already.

Retail has 2 features we do not:
1. **Event skipped.** If you take damage during an event before a fade out, you get kicked out of the event.
2. **De-aggro on fade to black.** If you make it to the fade out, mobs forget you and go home or depop.

Taking damage during a cs without being kicked out of that cs event is known to put the client in an invalid state like this. A relatively recent work around was done fro the Alexander fight, if I am not mistaken, because when it transitioned to the 2nd part people in range were agro'd before the event was done.

**Comment by Xaver-DaRed** [comment]
@RAIST5150 That happened in retail too iirc. mobs should not loose you on zoneline until you actually disapear.
As Teo said, we are lacking features for npc interaction

**Comment by RAIST5150** [comment]
True, but it seems to take considerably longer here before you actually leave a zone.  Can sometimes here several attack rounds fire off... more hits taken than I recall from retail.  Not talking the odd multihit move that might catch  you off guard, but regular attack rounds.

Just something I've learned to live with and guard against.. but may be a bit much for some, especially those new to everything.

**Comment by TeoTwawki** [comment]
for zonage on retail I learned to auto run before hitting the line - my side may fade to black but server side the character gets in a few more steps, buying precious milliseconds before the mob catches my squishy self. its no guarantee I live, but if I have to zone a mob I do it.
